js/interface.js

```diff
@@ -0,0 +1,64 @@
+var appDataSources = [];
+Fliplet.DataSources.get({ appId: Fliplet.Env.get('masterAppId'), attributes: ['id', 'name'] }).then(function (DSs) {
+  appDataSources = DSs.map(el => {
+    return { value: el.id, label: el.name };
+  });
+  Fliplet.Widget.generateInterface({
```

Comment:
You're calling `Fliplet.DataSources.get()` at the same time as `Fliplet.Widget.generateInterface()`. There's therefore no guarantee that `appDataSources` will be ready in time.